metadata-ingestion/src/datahub/ingestion/source/csv_enricher.py

```diff
@@ -264,28 +275,73 @@ def get_resource_description_work_unit(
+
+        entityClass = {
+            "chart": EditableChartPropertiesClass,
+            "dataset": EditableDatasetPropertiesClass,
+            "container": EditableContainerPropertiesClass,
+            "mlPrimaryKey": EditableMLPrimaryKeyPropertiesClass,
+            "mlModel": EditableMLModelPropertiesClass,
+            "mlModelGroup": EditableMLModelGroupPropertiesClass,
+            "mlFeatureTable": EditableMLFeatureTablePropertiesClass,
+            "mlFeature": EditableMLFeaturePropertiesClass,
+            "dashboard": EditableDashboardPropertiesClass,
+            "datajob": EditableDataJobPropertiesClass,
+            "dataflow": EditableDataFlowPropertiesClass,
+            "notebook": EditableNotebookPropertiesClass,
+        }.get(entityType, None)
+
+        if not entityClass:
+            raise ValueError(
+                f"Entity Type {entityType} cannot be operated on using csv-enricher"
+            )
+        current_editable_properties: Optional[
+            Union[
+                EditableDatasetPropertiesClass,
+                EditableContainerPropertiesClass,
+                EditableChartPropertiesClass,
+                EditableMLPrimaryKeyPropertiesClass,
+                EditableMLModelPropertiesClass,
+                EditableMLModelGroupPropertiesClass,
+                EditableMLFeatureTablePropertiesClass,
+                EditableMLFeaturePropertiesClass,
+                EditableDashboardPropertiesClass,
+                EditableDataJobPropertiesClass,
+                EditableDataFlowPropertiesClass,
+                EditableNotebookPropertiesClass,
+            ]
+        ] = None
         if self.ctx.graph and not self.should_overwrite:
             # Get the existing editable properties for the entity from the DataHub graph
             current_editable_properties = self.ctx.graph.get_aspect(
                 entity_urn=entity_urn,
-                aspect_type=EditableDatasetPropertiesClass,
+                aspect_type=entityClass,
             )
 
         if not current_editable_properties:
             # If we want to overwrite or there are no existing editable dataset properties, create a new object
-            current_editable_properties = EditableDatasetPropertiesClass(
-                created=get_audit_stamp(),
-                lastModified=get_audit_stamp(),
+            current_editable_properties = entityClass(
                 description=description,
             )
         else:
             current_editable_properties.description = description
+
+        # as not all classes have the same attributes
+        additionalAttributesClass = (
+            EditableDatasetPropertiesClass,
+            EditableChartPropertiesClass,
+            EditableDashboardPropertiesClass,
+            EditableDataJobPropertiesClass,
+            EditableDataFlowPropertiesClass,
+            EditableNotebookPropertiesClass,
+        )
```

Comment:
we can do this without hardcoding by using a hasattr check

```py
from datahub.metadata.schema_classes import (
    EditableDatasetPropertiesClass,
    EditableMLModelGroupPropertiesClass,
)

print(hasattr(EditableDatasetPropertiesClass, "created"))
print(hasattr(EditableMLModelGroupPropertiesClass, "created"))
```

Reply:
mypy really hates this one trick! :)